Revert
https://github.com/qt/qtbase/commit/c7eb2c173e79f2936985cd7589347e59b6005c7c
because it breaks the dialog: The font never updates and always returns the
initial font.

--- qtbase/src/plugins/platforms/cocoa/qcocoafontdialoghelper.mm.orig	2018-06-15 09:29:31.000000000 +0200
+++ qtbase/src/plugins/platforms/cocoa/qcocoafontdialoghelper.mm	2018-08-03 14:49:38.000000000 +0200
@@ -73,6 +74,8 @@
     return newFont;
 }
 
+@class QT_MANGLE_NAMESPACE(QNSFontPanelDelegate);
+
 @interface QT_MANGLE_NAMESPACE(QNSFontPanelDelegate) : NSObject<NSWindowDelegate, QNSPanelDelegate>
 {
     @public
@@ -109,8 +112,8 @@
     [mFontPanel setRestorable:NO];
     [mFontPanel setDelegate:self];
 
-    [NSFontManager sharedFontManager].target = self; // Action is changeFont:
+    [[NSFontManager sharedFontManager] setDelegate:self];
 
     [mFontPanel retain];
     return self;
 }
@@ -119,7 +123,7 @@
 {
     [mStolenContentView release];
     [mFontPanel setDelegate:nil];
-    [NSFontManager sharedFontManager].target = nil;
+    [[NSFontManager sharedFontManager] setDelegate:nil];
     [[NSNotificationCenter defaultCenter] removeObserver:self];
 
     [super dealloc];
